pkg/nodeconfig/nodeconfig.go

```diff
@@ -588,20 +587,38 @@ func (nc *nodeConfig) UpdateKubeletClientCA(contents []byte) error {
 	return nil
 }
 
-// ensureTrustedCABundle gets the trusted CA ConfigMap and ensures the cert bundle on the instance has up-to-date data
-func (nc *nodeConfig) ensureTrustedCABundle() error {
-	trustedCA := &core.ConfigMap{}
+// SyncTrustedCABundle builds the trusted CA ConfigMap from image registry certificates and the proxy trust bundle
+// and ensures the cert bundle on the instance has up-to-date data
+func (nc *nodeConfig) SyncTrustedCABundle() error {
+	caBundle := ""
+	var cc mcfg.ControllerConfig
 	if err := nc.client.Get(context.TODO(), types.NamespacedName{Namespace: nc.wmcoNamespace,
-		Name: certificates.ProxyCertsConfigMap}, trustedCA); err != nil {
-		return fmt.Errorf("unable to get ConfigMap %s: %w", certificates.ProxyCertsConfigMap, err)
+		Name: "machine-config-controller"}, &cc); err != nil {
+		return err
+	}
+	for _, bundle := range cc.Spec.ImageRegistryBundleUserData {
+		// Accumulate the certs into a single bundle, adding a comment with the image repo name for observability
+		caBundle += fmt.Sprintf("# %s\n%s\n\n", strings.ReplaceAll(bundle.File, "..", ":"), bundle.Data)
```

Comment:
could this caBundle be moved to a function instead of duplicating the code?

Reply:
+1

and consider adding unit test coverage.